### PR TITLE
Fix issue where checkbox remained disabled when credit card was disabled

### LINKED ISSUE #5525: Can't enable CC payments
### Describe the bug
As seen in 5934805-zen the user let us know that they were unable to enable credit card payments.

The checkbox was not ticked but was disabled, i.e. the appearance reflects that CC payments are not enabled, and the checkbox can not be checked to enable.

### To Reproduce

1. Disable UPE
2. Uncheck CC
3. Enable UPE but don't complete the setup. Close your browser tab when you see this: 
4. Navigate back to the WCPay settings. CC is disabled and can not enable.

### Expected behavior
To be able to enable CC payments if they're not already.

### Desktop (please complete the following information):

* OS: [e.g. iOS]
* Browser [e.g. chrome, safari]
* Version [e.g. 22]

### Smartphone (please complete the following information):

* Device: [e.g. iPhone6]
* OS: [e.g. iOS8.1]
* Browser [e.g. stock browser, safari]
* Version [e.g. 22]

### PR DESCRIPTION
**Pull request by zmaglica**
Fixes #5525 

#### Changes proposed in this Pull Request

Resolves an issue encountered in the environment where UPE and Credit Card payment method is disabled. Previously, enabling UPE after disabling the credit card resulted in the inability to click on the credit card checkbox.

#### Testing instructions

1. Disable UPE
2. Uncheck CC
3. Enable UPE but don't complete the setup. Close your browser tab when you see this: 
4. Navigate back to the WCPay settings. Make sure that you can enable Credit Card payment method.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `6504` or branch name `fix/5525/cant-enable-cc-payments` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=fix/5525/cant-enable-cc-payments) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 93c89df5773c0543b7d6c3329106553128284e06
- Build time: 2023-06-20 14:35:07 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/payment-methods/index.js [hunk 1]
```diff
@@ -125,6 +125,8 @@ const PaymentMethods = () => {
 		.filter( ( id ) => 'link' !== id )
 		.map( ( methodId ) => methodsConfiguration[ methodId ] );
 
+	const isCreditCardEnabled = enabledMethodIds.includes( 'card' );
```
Comment (RadoslavGeorgiev) [inline client/payment-methods/index.js:153]: @zmaglica the code works as advertised, but I'd like some context included about this line:

1. Is `enabledMethodIds` UPE-specific?
2. Is it the same with split-UPE?
3. Does `enabledPaymentMethodIds` represent available ones, or does UPE force-enable it here?

File: changelog/fix-5525-cant-enable-cc-payments [hunk 2]
```diff
@@ -0,0 +1,4 @@
+Significance: patch
+Type: fix
+
+Corrected bug where checkbox remained disabled when credit card was disabled
```
Comment (RadoslavGeorgiev) [inline changelog/fix-5525-cant-enable-cc-payments]: People who need this bug fixed will probably have sufficient context, but let's consider whether this message conveys all the needed information. Should it maybe mention the settings (page)? Perhaps replace "remained disabled" with "could not be enabled" or something similar?

**Review by RadoslavGeorgiev — CHANGES_REQUESTED** [review]

File: client/payment-methods/index.js (hunk 1 above)
Comment (zmaglica) [reply client/payment-methods/index.js:153]: I have tested both with UPE and Split-UPE replicating the steps above and the error was not presented anymore. I think that the `enabledPaymentMethodIds` represents the ones that are checked or selected. The available ones can be found in `availableMethods` or `useGetAvailablePaymentMethodIds()`

File: changelog/fix-5525-cant-enable-cc-payments (hunk 2 above)
Comment (zmaglica) [reply changelog/fix-5525-cant-enable-cc-payments]: Good point. Fixed in https://github.com/Automattic/woocommerce-payments/pull/6504/commits/b4b286fd3a20da418ce5646f445fe46fa6a45ccd

**Review by RadoslavGeorgiev — APPROVED** [review]
Great work, changes LGTM! :shipit:

**Pull request merged**